domehunter/dome_control.py

```diff
@@ -355,81 +386,176 @@ def calibrate_dome_encoder_counts(self, num_cal_rotations=2):
-        while rotation_count < num_cal_rotations:
-            self._rotate_dome(Direction.CW)
-            if self.testing:
-                # tell the fake home sensor that we have left home
-                self._home_sensor_pin.drive_low()
-                self._simulate_ticks(num_ticks=10)
-            self._home_sensor.wait_for_inactive(timeout=self.wait_timeout)
-            self._home_sensor.wait_for_active(timeout=self.wait_timeout)
-
-            if self.testing:
-                # tell the fake home sensor that we have come back to home
-                time.sleep(0.1)
-                self._home_sensor_pin.drive_high()
-                time.sleep(0.1)
-
-            # without this pause the wait_for_active wasn't waiting (???)
-            time.sleep(0.1)
-
-            rotation_count += 1
-
-        self._stop_moving()
-
-        # set the azimuth per encoder tick factor based on how many ticks we
-        # counted over n rotations
-        self._degrees_per_tick = Angle(
-            360 / (self.encoder_count / rotation_count) * u.deg)
-        # update the az_position_tolerance
-        self.az_position_tolerance = max(
-            self.az_position_tolerance, 2 * self.degrees_per_tick)
-        self.calibrating = False
+        logger.debug(f'calibrate_dome_encoder_counts: Starting calibration rotations.')
+        self._move_event.set()
+        self._rotate_dome(Direction.CW)
+        self._calibrating = True
+        calibrate = threading.Thread(target=self._calibrate)
+        cal_monitor = threading.Thread(target=self._thread_condition,
+                                       args=(self._calibration_completed,))
+        calibrate.start()
+        cal_monitor.start()
 
     def find_home(self):
         """
         Move Dome to home position.
 
         """
+        if self._move_event.is_set():
+            logger.debug('Movement command in progress')
+            raise Exception("Movement command in progress")
+        # iniate the movement and set the _move_event flag
+        logger.debug(f'find_home: Finding Home')
+        self._move_event.set()
         self._rotate_dome(Direction.CW)
         time.sleep(0.1)
-        self._home_sensor.wait_for_active(timeout=self.wait_timeout)
+        homing = threading.Thread(target=self._thread_condition,
+                                  args=(self._find_home_condition,))
+        homing.start()
         if self.testing:
             # in testing mode need to "fake" the activation of the home pin
-            time.sleep(0.5)
-            self._home_sensor_pin.drive_high()
-
-        self._stop_moving()
+            home_pin_high = threading.Timer(0.5,
+                                            self._home_sensor_pin.drive_high)
+            home_pin_high.start()
 
     def sync(self, az):
+        """
+        Sync encoder count to azimuth position.
+        Parameters
+        ----------
+        az : float
+            Desired dome azimuth position in degrees.
```

Comment:
Technically this is the current dome azimuth in degrees, not desired dome azimuth. The purpose of this function is to tell the dome controller what the actual dome azimuth is, and have it adjust its encoder count to match. It's an alternative to homing.